### ISSUE TITLE
ClassCastException: org.bson.LazyBSONList cannot be cast to org.bson.types.BasicBSONList

### ISSUE THREAD

**Issue by hlassiege** [clun/ff4j#28]
When adding a feature using FeatureStoreMongodb, the produced json in database is : 

```
> db.ff4j.find().pretty()
{
        "_id" : "choose_lang",
        "enable" : false,
        "description" : "Capability to choose language",
        "groupname" : null,
        "strategy" : "com.hopwork.ff4j.strategy.AdminOrAccountFilterStrategy",
        "expression" : "some values",
        "roles" : [ ]
}
```

However, it fails in FeatureDBObjectMapper on line 92 : 

```
        if (dbObject.containsField(ROLES)) {
            for (Object role : (BasicBSONList) dbObject.get(ROLES)) {
                authorisation.add(role.toString());
            }
        }

=> 
ClassCastException: org.bson.LazyBSONList cannot be cast to org.bson.types.BasicBSONList
```

If I unset this variable in mongoshell, everything works correctly.

**Comment by clun** [comment]
Hum... is it a trick from the mongodb-driver returning different BSONList if the list is empty ? I'm pretty sure to test this behaviour if roles are set. 

Well I would like to cast to the common upper class "BSONObject" but it's not iterable.... I will have to use the ugly (instanceof ?) beurk...

**Comment by clun** [comment]
Well, cannot reproduce this one (Mongodb v2.6.4, MongoDriver v2.12.2)

> db.feature.find({"_id":"AwesomeFeature"}).pretty();
```
{
	"_id" : "AwesomeFeature",
	"enable" : true,
	"description" : "some desc",
	"groupname" : null,
	"strategy" : null,
	"expression" : null,
	"roles" : [ ]
}
```
Test passing (no exception at least)
```
// Given
 MongoCredential credential = MongoCredential.createMongoCRCredential("xx", "yyy", "xxx".toCharArray());
ServerAddress adr = new ServerAddress("localhost", 22012);
MongoClient mongoClient = new MongoClient(adr, Arrays.asList(credential));
DBCollection features = mongoClient.getDB( "FF4J" ).getCollection("feature");
// When
FeatureStore mongoStore = new FeatureStoreMongoDB(features, "ff4j.xml");
Feature f = mongoStore.read("AwesomeFeature");
```
still investigate...

**Comment by hlassiege** [comment]
Hum... I'm using the same versions

I made a little patch for my purpose and I just check the existence of the field in mapAuthorization method (FeatureDBObjectMapper.java)

```
        if (dbObject.containsField(ROLES)) {
            for (Object role : (Iterable) dbObject.get(ROLES)) {
                authorisation.add(role.toString());
            }
        }
```
Could you check with a null value instead of an empty value ?

**Comment by clun** [comment]
I also test with the null value and same result. The test for existence was already there but casting as Iterable is a good enough solution work in all cases. I will go this way.

**Comment by clun** [comment]
Corrected in 1.3.1 with the iterable trick
